lib/Containers/MerkleTree.cpp

```diff
@@ -474,16 +461,67 @@ std::unique_ptr<MerkleTree<BranchingBits, LockStripes>> MerkleTree<BranchingBits
       new MerkleTree<BranchingBits, LockStripes>(*this));
 }
 
+template <std::size_t const BranchingBits, std::size_t const LockStripes>
+std::unique_ptr<MerkleTree<BranchingBits, LockStripes>>
+MerkleTree<BranchingBits, LockStripes>::cloneWithDepth(std::size_t newDepth) {
+  std::shared_lock<std::shared_mutex> guard(_bufferLock);
+
+  if (newDepth == meta().maxDepth) {
+    return std::unique_ptr<MerkleTree<BranchingBits, LockStripes>>(
+        new MerkleTree<BranchingBits, LockStripes>(*this));
```

Comment:
```suggestion
    return std::make_unique<MerkleTree<BranchingBits, LockStripes>>(*this);
```

Reply:
`make_unique` requires the constructor to be public, while this one is currently protected. We don't really want that constructor to part of the public API, so I'm going to leave this as-is.